bot/cogs/regular_expressions.py

```diff
@@ -0,0 +1,87 @@
+import re
+from typing import Iterable, Optional
+
+from discord.ext.commands import BadArgument, Cog, Context, group
+
+from bot.bot import Bot
+
+
+def format_error(e: re.error) -> Iterable[str]:
+    r"""
+    Format a regexp parsing error message to display in a response.
+
+    \w+(
+    ^
+    missing ), unterminated subpattern
+    """
+    line_with_regexp = e.pattern
+    line_with_caret = " "*e.pos + "^"
+    line_with_explanation = e.msg
+    return [line_with_regexp, line_with_caret, line_with_explanation]
+
+
+def format_match(match: Optional[re.Match]) -> Iterable[str]:
+    """
+    Format a match result to display in a response.
+
+    Example:
+        hello1 23 456world
+    0:      ^^^^^^^^
+    2:           ^^^
+    """
+    if match is None:
+        return ["No match"]
+    else:
+        group_carets = []
+        captured_positions = match.regs
+
+        for group_index, (start, end) in enumerate(captured_positions):
+            group_is_missing = (start, end) == (-1, -1)
+            if not group_is_missing:
+                group_carets.append(
+                    f"{group_index:>2}: " + " "*start + "^"*(end - start)
+                )
+
+        return ["    " + match.string, *group_carets]
+
+
+def convert_regexp(supposed_regexp: str) -> re.Pattern:
+    """
+    Argument converter.
+
+    Attempts to interpret the given string as a regexp
+    and returns a compiled pattern as a result.
+    """
+    try:
+        return re.compile(supposed_regexp)
+    except re.error as e:
+        error_lines = "\n".join(format_error(e))
+        raise BadArgument(
+            "Syntax error in a regular expression: "
+            "```\n"
+            + error_lines
+            + "\n```"
+        )
+
+
+class RegularExpressions(Cog):
+    """Commands related to regular expressions."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @group(name='regexp', aliases=('regex', 're'), invoke_without_command=True)
+    async def regexp_group(self, ctx: Context) -> None:
+        """Commands for exploring the misterious worldof regular expressions."""
```

Comment:
Minor typos here.
```suggestion
        """Commands for exploring the mysterious world of regular expressions."""
```

Reply:
Thanks, I'll fix that. I think `format_*` docstrings can be improved as well.